app/models/cluster_group.rb

```diff
@@ -40,6 +40,8 @@ def by_name_and_type(name, type, subsample_threshold=nil)
   # fixed values to subsample at
   SUBSAMPLE_THRESHOLDS = [1000, 10000, 20000].freeze
 
+  MAX_THRESHOLD = 250000
```

Comment:
How did you come up with this number?

Reply:
Great question!  Estimation.  We know the search fails with 1.3 million cells, so I figured 25% of that would be safe.

Further testing reveals that 250K cells throws the same `Maximum call stack size exceeded` error on production in `createTracesAndLayout`.  This PR's optimization in `arrayMin` and `arrayMax` should mitigate that; it enabled me to get past the previous line of failure with the 1.3M-cell study, but too slowly for decent UX.  The current UI cannot handle gene search results for a 153,539-cell study, but can for a 118,861-cell study.

So I've lowered `MAX_THRESHOLD` to 100,000 cells.  This is 5x our highest enumerated threshold of 20,000 cells, and comfortably below our empirically verified limit.  More substantial performance engineering this quarter and next should allow to us to raise the threshold.

Reply:
I think this is a really good plan.